[WIP] Adding dashboard view with quotas to Overview documentation

bmignano opened the pull request:
Adding screenshot for dashboard view on Overview / Status page. Will update after https://github.com/openshift/openshift-origin-design/pull/131.

FYI @openshift/team-ux-review 

![status-page](https://user-images.githubusercontent.com/40238481/48736166-859f9b00-ec18-11e8-9c3a-8c76bd83f9ca.png)

spadgett commented:
If there are an odd number of quota objects, should the last one stretch to 100%?

Should we include quota objects that don't have any CPU or memory values set?

beanh66 commented:
I vote yes to stretching to 100% when there's an odd number and no to showing for objects that don't have values set. Don't feel too strongly though so open to other ideas.

bmignano commented:
Agreed with stretching the last object to 100% if there's an odd number.

Not sure about hiding quota objects that don't have values set. I worry it could be misleading if a user has 3 quota resources but only 2 show up on the dashboard and there's no clear explanation why. I'd have to think more on what a good solution would be though..

dtaylor113 commented:
> Should we include quota objects that don't have any CPU or memory values set?

Can there be Scopes but no CPU or memory values set?  Ie. Scopes but no quota gauge charts?

bmignano commented:
@dtaylor113 I was wondering the same thing. I think it's also possible for a quota to have limits on the number of resources you can create, but maybe not have CPU or memory values or scopes. Would we want to show those resource quotas? I think it'd be valuable to figure out which parts of the resource quota are important to surface on the overview so we can account for as many cases as possible. @spadgett thoughts?

dtaylor113 commented:
Hi @bmignano, do we really want the gauge chart labels to be on the bottom?  In other implementations, the labels by default on on the top: https://github.com/openshift/openshift-origin-design/blob/392e7631f59f031109b7d47a09cce5357019fa8a/web-console/4.0-designs/resource-quota/resource-quota.md

bmignano commented:
@dtaylor113 I noticed that too - I switched them in my updated mocks but have not pushed changes yet! Thanks for pointing that out!

serenamarie125 commented:
If there are an odd number I don't think it should stretch, I'd leave a blank space

beanh66 commented:
After speaking with @bmignano it sounds like we won't show the empty state of a quota and @serenamarie125 agreed, if there is an odd number we would do something like this: 
![1](https://user-images.githubusercontent.com/18057044/49541246-41f79300-f8a0-11e8-9d22-5a5f5c166dd4.png)

FYI @spadgett

bmignano commented:
@serenamarie125 @beanh66 I split the status page into two separate docs in the README (Overview and Dashboard). Also added the updated resource quota guidelines to the dashboard documentation.

dtaylor113 commented:
> Can there be Scopes but no CPU or memory values set? Ie. Scopes but no quota gauge charts?

Still curious about ^^

dtaylor113 commented:
So object-count quotas will not have any quota gauge charts, but we do show on the quota details page:
![image](https://user-images.githubusercontent.com/12733153/49544379-b9312500-f8a8-11e8-8680-921847191783.png)
Any thoughts on putting something like the above table in a quota box/group? --maybe just the first two columns (name & capacity icon)?

beanh66 commented:
@dtaylor113 I think based on our meetings with @spadgett, just the CPU and Memory gauges were prioritized for the dashboard view, but users can click on a resource quota name from this view to see the detail view which does include a table like the one you're suggesting, doesn't it?

dtaylor113 commented:
> @dtaylor113 I think based on our meetings with @spadgett, just the CPU and Memory gauges were prioritized for the dashboard view, but users can click on a resource quota name from this view to see the detail view which does include a table like the one you're suggesting, doesn't it?

Object Count quotas will not show up in the dashboard because they do not have CPU and Memory gauges, nor (I believe) Scopes, so we decided not to show quotas in the dashboard which do not have these set (the 'empty state' discussions above).  I was suggesting maybe that table should be shown in the dashboard so Object Count quotas get some representation, but maybe it's not vital enough to be in the dashboard.

beanh66 commented:
I think @bmignano had a mock with that option. Instead of empty state, showing the table of info. I guess it didn't sound like a priority yet but up to you and @spadgett!
